crates/nu-cli/tests/completions/mod.rs

```diff
@@ -357,6 +357,24 @@ fn custom_completions_override_span() {
     "foo --test", None,
     vec!["--test".into(), "foo --test bar".into()]
 )]
+/// Flag value completion for directories
+#[case::flag_value_and_subcommands(r#"
+    def foo [--test: directory] {}
```

Comment:
I'd like a test case with another positional argument or flag before the one being completed, just for peace of mind

Reply:
That's a great idea, actually this PR fixes a bug of wrong positional arg index for the new arg value  dynamic completion. I'll add one for that as well.